collections-contrib/src/main/scala/strawman/collection/immutable/MultiSet.scala

```diff
@@ -10,10 +10,38 @@ import strawman.collection.decorators.ImmutableMapDecorator
   * An immutable multiset
   * @tparam A the element type of the collection
   */
-class MultiSet[A] private (elems: Map[A, Int])
+trait MultiSet[A]
   extends collection.MultiSet[A]
     with Iterable[A]
-    with collection.MultiSetOps[A, MultiSet, MultiSet[A]] {
+    with MultiSetOps[A, MultiSet, MultiSet[A]] {
+
+  override def iterableFactory: IterableFactory[MultiSet] = MultiSet
+}
```

Comment:
Can you elaborate on the problem that is solved by introducing the separation `MultiSet`/`MultiSetImpl`?

Reply:
Oh maybe that’s for making `SortedMultiSet` a subtype of `MultiSet`?

Reply:
Yes, you want the usual diamond pattern with all of:

- mutable.SortedMultiSet <: collection.SortedMultiSet
- immutable.SortedMultiSet <: collection.SortedMultiSet
- mutable.MultiSet <: collection.MultiSet
- immutable.MultiSet <: collection.MultiSet

- collection.SortedMultiSet <: collection.MultiSet
- mutable.SortedMultiSet <: mutable.MultiSet
- immutable.SortedMultiSet <: immutable.MultiSet

This requires that `MultiSet` is a trait.